tiledb/api/src/error.rs

```diff
@@ -98,4 +100,13 @@ impl Drop for Error {
     }
 }
 
+impl Serialize for Error {
```

Comment:
Arguably this is questonable, but it makes the `json!` in the `Debug` impls way more concise.  Although maybe a generic function would work instead actuallly...

Reply:
I'm only confused why you think its questionable. I mean, we could get fancier with making it an object with richer information, but as far ask I can tell there's no worry about somehow causing a panic.

Reply:
This is how I want to format the error messages when they would show up in `Debug`, but that doesn't mean it's how we should serialize an error message.  Whereas this conflates the two.